<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
"http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd" [
<!ENTITY mdash  "&#x2014;" >
<!ENTITY % version SYSTEM "version.ent">
%version;
]>
<!--
     This XML document is generated using the system_messages.py tool
     based on the .mes message files.

     Do not edit this file.
-->
<book>
  <?xml-stylesheet href="bind10-guide.css" type="text/css"?>

  <bookinfo>
    <title>BIND 10 Messages Manual</title>

    <copyright>
      <year>2011</year><holder>Internet Systems Consortium, Inc.</holder>
    </copyright>

    <abstract>
      <para>BIND 10 is a Domain Name System (DNS) suite managed by
	  Internet Systems Consortium (ISC). It includes DNS libraries
	  and modular components for controlling authoritative and
	  recursive DNS servers.
      </para>
      <para>
        This is the messages manual for BIND 10 version &__VERSION__;.
	    The most up-to-date version of this document, along with
	    other documents for BIND 10, can be found at
        <ulink url="http://bind10.isc.org/docs"/>.
      </para>
    </abstract>

    <releaseinfo>This is the messages manual for BIND 10 version
        &__VERSION__;.</releaseinfo>
  </bookinfo>

  <chapter id="intro">
    <title>Introduction</title>
    <para>
      This document lists each message that can be logged by the
      programs in the BIND 10 package.  Each entry in this manual
      is of the form:
      <screen>IDENTIFICATION message-text</screen>
      ... where "IDENTIFICATION" is the message identification included
      in each message logged and "message-text" is the accompanying
      message text.  The "message-text" may include placeholders of the
      form "%1", "%2" etc.; these parameters are replaced by relevant
      values when the message is logged.
    </para>
    <para>
      Each entry is also accompanied by a description giving more
      information about the circumstances that result in the message
      being logged.
    </para>
    <para>
      For information on configuring and using BIND 10 logging,
      refer to the <ulink url="bind10-guide.html">BIND 10 Guide</ulink>.
    </para>
  </chapter>

  <chapter id="messages">
    <title>BIND 10 Messages</title>
    <para>
      <variablelist>

<varlistentry id="ASIODNS_FETCH_COMPLETED">
<term>ASIODNS_FETCH_COMPLETED upstream fetch to %1(%2) has now completed</term>
<listitem><para>
A debug message, this records that the upstream fetch (a query made by the
resolver on behalf of its client) to the specified address has completed.
</para></listitem>
</varlistentry>

<varlistentry id="ASIODNS_FETCH_STOPPED">
<term>ASIODNS_FETCH_STOPPED upstream fetch to %1(%2) has been stopped</term>
<listitem><para>
An external component has requested the halting of an upstream fetch.  This
is an allowed operation, and the message should only appear if debug is
enabled.
</para></listitem>
</varlistentry>

<varlistentry id="ASIODNS_OPEN_SOCKET">
<term>ASIODNS_OPEN_SOCKET error %1 opening %2 socket to %3(%4)</term>
<listitem><para>
The asynchronous I/O code encountered an error when trying to open a socket
of the specified protocol in order to send a message to the target address.
The number of the system error that cause the problem is given in the
message.
</para></listitem>
</varlistentry>

<varlistentry id="ASIODNS_READ_DATA">
<term>ASIODNS_READ_DATA error %1 reading %2 data from %3(%4)</term>
<listitem><para>
The asynchronous I/O code encountered an error when trying to read data from
the specified address on the given protocol.  The number of the system
error that cause the problem is given in the message.
</para></listitem>
</varlistentry>

<varlistentry id="ASIODNS_READ_TIMEOUT">
<term>ASIODNS_READ_TIMEOUT receive timeout while waiting for data from %1(%2)</term>
<listitem><para>
An upstream fetch from the specified address timed out.  This may happen for
any number of reasons and is most probably a problem at the remote server
or a problem on the network.  The message will only appear if debug is
enabled.
</para></listitem>
</varlistentry>

<varlistentry id="ASIODNS_SEND_DATA">
<term>ASIODNS_SEND_DATA error %1 sending data using %2 to %3(%4)</term>
<listitem><para>
The asynchronous I/O code encountered an error when trying send data to
the specified address on the given protocol.  The the number of the system
error that cause the problem is given in the message.
</para></listitem>
</varlistentry>

<varlistentry id="ASIODNS_UNKNOWN_ORIGIN">
<term>ASIODNS_UNKNOWN_ORIGIN unknown origin for ASIO error code %1 (protocol: %2, address %3)</term>
<listitem><para>
An internal consistency check on the origin of a message from the
asynchronous I/O module failed. This may indicate an internal error;
please submit a bug report.
</para></listitem>
</varlistentry>

<varlistentry id="ASIODNS_UNKNOWN_RESULT">
<term>ASIODNS_UNKNOWN_RESULT unknown result (%1) when IOFetch::stop() was executed for I/O to %2(%3)</term>
<listitem><para>
An internal error indicating that the termination method of the resolver's
upstream fetch class was called with an unknown result code (which is
given in the message).  Please submit a bug report.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_AXFR_ERROR">
<term>AUTH_AXFR_ERROR error handling AXFR request: %1</term>
<listitem><para>
This is a debug message produced by the authoritative server when it
has encountered an error processing an AXFR request. The message gives
the reason for the error, and the server will return a SERVFAIL code to
the sender.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_AXFR_UDP">
<term>AUTH_AXFR_UDP AXFR query received over UDP</term>
<listitem><para>
This is a debug message output when the authoritative server has received
an AXFR query over UDP. Use of UDP for AXFRs is not permitted by the
protocol, so the server will return a FORMERR error to the sender.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_COMMAND_FAILED">
<term>AUTH_COMMAND_FAILED execution of command channel instruction '%1' failed: %2</term>
<listitem><para>
Execution of the specified command by the authoritative server failed. The
message contains the reason for the failure.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_CONFIG_CHANNEL_CREATED">
<term>AUTH_CONFIG_CHANNEL_CREATED configuration session channel created</term>
<listitem><para>
This is a debug message indicating that authoritative server has created
the channel to the configuration manager.  It is issued during server
startup is an indication that the initialization is proceeding normally.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_CONFIG_CHANNEL_ESTABLISHED">
<term>AUTH_CONFIG_CHANNEL_ESTABLISHED configuration session channel established</term>
<listitem><para>
This is a debug message indicating that authoritative server
has established communication the configuration manager over the
previously-created channel. It is issued during server startup is an
indication that the initialization is proceeding normally.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_CONFIG_CHANNEL_STARTED">
<term>AUTH_CONFIG_CHANNEL_STARTED configuration session channel started</term>
<listitem><para>
This is a debug message, issued when the authoritative server has
posted a request to be notified when new configuration information is
available. It is issued during server startup is an indication that
the initialization is proceeding normally.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_CONFIG_LOAD_FAIL">
<term>AUTH_CONFIG_LOAD_FAIL load of configuration failed: %1</term>
<listitem><para>
An attempt to configure the server with information from the configuration
database during the startup sequence has failed. (The reason for
the failure is given in the message.) The server will continue its
initialization although it may not be configured in the desired way.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_CONFIG_UPDATE_FAIL">
<term>AUTH_CONFIG_UPDATE_FAIL update of configuration failed: %1</term>
<listitem><para>
At attempt to update the configuration the server with information
from the configuration database has failed, the reason being given in
the message.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_DATA_SOURCE">
<term>AUTH_DATA_SOURCE data source database file: %1</term>
<listitem><para>
This is a debug message produced by the authoritative server when it accesses a
datebase data source, listing the file that is being accessed.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_DNS_SERVICES_CREATED">
<term>AUTH_DNS_SERVICES_CREATED DNS services created</term>
<listitem><para>
This is a debug message indicating that the component that will handling
incoming queries for the authoritiative server (DNSServices) has been
successfully created. It is issued during server startup is an indication
that the initialization is proceeding normally.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_HEADER_PARSE_FAIL">
<term>AUTH_HEADER_PARSE_FAIL unable to parse header in received DNS packet: %1</term>
<listitem><para>
This is a debug message, generated by the authoritative server when an
attempt to parse the header of a received DNS packet has failed. (The
reason for the failure is given in the message.) The server will drop the
packet.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_LOAD_TSIG">
<term>AUTH_LOAD_TSIG loading TSIG keys</term>
<listitem><para>
This is a debug message indicating that the authoritiative server
has requested the keyring holding TSIG keys from the configuration
database. It is issued during server startup is an indication that the
initialization is proceeding normally.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_LOAD_ZONE">
<term>AUTH_LOAD_ZONE loaded zone %1/%2</term>
<listitem><para>
This debug message is issued during the processing of the 'loadzone' command
when the authoritative server has successfully loaded the named zone of the
named class.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_MEM_DATASRC_DISABLED">
<term>AUTH_MEM_DATASRC_DISABLED memory data source is disabled for class %1</term>
<listitem><para>
This is a debug message reporting that the authoritative server has
discovered that the memory data source is disabled for the given class.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_MEM_DATASRC_ENABLED">
<term>AUTH_MEM_DATASRC_ENABLED memory data source is enabled for class %1</term>
<listitem><para>
This is a debug message reporting that the authoritative server has
discovered that the memory data source is enabled for the given class.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_NOTIFY_QUESTIONS">
<term>AUTH_NOTIFY_QUESTIONS invalid number of questions (%1) in incoming NOTIFY</term>
<listitem><para>
This debug message is logged by the authoritative server when it receives
a NOTIFY packet that contains zero or more than one question. (A valid
NOTIFY packet contains one question.) The server will return a FORMERR
error to the sender.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_NOTIFY_RRTYPE">
<term>AUTH_NOTIFY_RRTYPE invalid question RR type (%1) in incoming NOTIFY</term>
<listitem><para>
This debug message is logged by the authoritative server when it receives
a NOTIFY packet that an RR type of something other than SOA in the
question section. (The RR type received is included in the message.) The
server will return a FORMERR error to the sender.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_NO_STATS_SESSION">
<term>AUTH_NO_STATS_SESSION session interface for statistics is not available</term>
<listitem><para>
The authoritative server had no session with the statistics module at the
time it attempted to send it data: the attempt has been abandoned. This
could be an error in configuration.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_NO_XFRIN">
<term>AUTH_NO_XFRIN received NOTIFY but XFRIN session is not running</term>
<listitem><para>
This is a debug message produced by the authoritative server when it receives
a NOTIFY packet but the XFRIN process is not running. The packet will be
dropped and nothing returned to the sender.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_PACKET_PARSE_ERROR">
<term>AUTH_PACKET_PARSE_ERROR unable to parse received DNS packet: %1</term>
<listitem><para>
This is a debug message, generated by the authoritative server when an
attempt to parse a received DNS packet has failed due to something other
than a protocol error. The reason for the failure is given in the message;
the server will return a SERVFAIL error code to the sender.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_PACKET_PROTOCOL_ERROR">
<term>AUTH_PACKET_PROTOCOL_ERROR DNS packet protocol error: %1. Returning %2</term>
<listitem><para>
This is a debug message, generated by the authoritative server when an
attempt to parse a received DNS packet has failed due to a protocol error.
The reason for the failure is given in the message, as is the error code
that will be returned to the sender.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_PACKET_RECEIVED">
<term>AUTH_PACKET_RECEIVED message received:\n%1</term>
<listitem><para>
This is a debug message output by the authoritative server when it
receives a valid DNS packet.
</para><para>
Note: This message includes the packet received, rendered in the form of
multiple lines of text. For this reason, it is suggested that this log message
not be routed to the syslog file, where the multiple lines could confuse
programs that expect a format of one message per line.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_PROCESS_FAIL">
<term>AUTH_PROCESS_FAIL message processing failure: %1</term>
<listitem><para>
This message is generated by the authoritative server when it has
encountered an internal error whilst processing a received packet:
the cause of the error is included in the message.
</para><para>
The server will return a SERVFAIL error code to the sender of the packet.
However, this message indicates a potential error in the server.
Please open a bug ticket for this issue.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_RECEIVED_COMMAND">
<term>AUTH_RECEIVED_COMMAND command '%1' received</term>
<listitem><para>
This is a debug message issued when the authoritative server has received
a command on the command channel.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_RECEIVED_SENDSTATS">
<term>AUTH_RECEIVED_SENDSTATS command 'sendstats' received</term>
<listitem><para>
This is a debug message issued when the authoritative server has received
a command from the statistics module to send it data. The 'sendstats'
command is handled differently to other commands, which is why the debug
message associated with it has its own code.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_RESPONSE_RECEIVED">
<term>AUTH_RESPONSE_RECEIVED received response message, ignoring</term>
<listitem><para>
This is a debug message, this is output if the authoritative server
receives a DNS packet with the QR bit set, i.e. a DNS response. The
server ignores the packet as it only responds to question packets.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_SEND_ERROR_RESPONSE">
<term>AUTH_SEND_ERROR_RESPONSE sending an error response (%1 bytes):\n%2</term>
<listitem><para>
This is a debug message recording that the authoritative server is sending
an error response to the originator of the query. A previous message will
have recorded details of the failure.
</para><para>
Note: This message includes the packet sent, rendered in the form of
multiple lines of text. For this reason, it is suggested that this log message
not be routed to the syslog file, where the multiple lines could confuse
programs that expect a format of one message per line.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_SEND_NORMAL_RESPONSE">
<term>AUTH_SEND_NORMAL_RESPONSE sending an error response (%1 bytes):\n%2</term>
<listitem><para>
This is a debug message recording that the authoritative server is sending
a response to the originator of a query.
</para><para>
Note: This message includes the packet sent, rendered in the form of
multiple lines of text. For this reason, it is suggested that this log message
not be routed to the syslog file, where the multiple lines could confuse
programs that expect a format of one message per line.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_SERVER_CREATED">
<term>AUTH_SERVER_CREATED server created</term>
<listitem><para>
An informational message indicating that the authoritative server process has
been created and is initializing. The AUTH_SERVER_STARTED message will be
output when initialization has successfully completed and the server starts
accepting queries.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_SERVER_FAILED">
<term>AUTH_SERVER_FAILED server failed: %1</term>
<listitem><para>
The authoritative server has encountered a fatal error and is terminating. The
reason for the failure is included in the message.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_SERVER_STARTED">
<term>AUTH_SERVER_STARTED server started</term>
<listitem><para>
Initialization of the authoritative server has completed successfully
and it is entering the main loop, waiting for queries to arrive.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_SQLITE3">
<term>AUTH_SQLITE3 nothing to do for loading sqlite3</term>
<listitem><para>
This is a debug message indicating that the authoritative server has
found that the data source it is loading is an SQLite3 data source,
so no further validation is needed.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_STATS_CHANNEL_CREATED">
<term>AUTH_STATS_CHANNEL_CREATED STATS session channel created</term>
<listitem><para>
This is a debug message indicating that the authoritative server has
created a channel to the statistics process.  It is issued during server
startup is an indication that the initialization is proceeding normally.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_STATS_CHANNEL_ESTABLISHED">
<term>AUTH_STATS_CHANNEL_ESTABLISHED STATS session channel established</term>
<listitem><para>
This is a debug message indicating that the authoritative server
has established communication over the previously created statistics
channel.  It is issued during server startup is an indication that the
initialization is proceeding normally.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_STATS_COMMS">
<term>AUTH_STATS_COMMS communication error in sending statistics data: %1</term>
<listitem><para>
An error was encountered when the authoritiative server tried to send data
to the statistics daemon. The message includes additional information
describing the reason for the failure.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_STATS_TIMEOUT">
<term>AUTH_STATS_TIMEOUT timeout while sending statistics data: %1</term>
<listitem><para>
The authoritative server sent data to the statistics daemon but received
no acknowledgement within the specified time. The message includes
additional information describing the reason for the failure.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_STATS_TIMER_DISABLED">
<term>AUTH_STATS_TIMER_DISABLED statistics timer has been disabled</term>
<listitem><para>
This is a debug message indicating that the statistics timer has been
disabled in the authoritative server and no statistics information is
being produced.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_STATS_TIMER_SET">
<term>AUTH_STATS_TIMER_SET statistics timer set to %1 second(s)</term>
<listitem><para>
This is a debug message indicating that the statistics timer has been
enabled and that the authoritative server will produce statistics data
at the specified interval.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_UNSUPPORTED_OPCODE">
<term>AUTH_UNSUPPORTED_OPCODE unsupported opcode: %1</term>
<listitem><para>
This is a debug message, produced when a received DNS packet being
processed by the authoritative server has been found to contain an
unsupported opcode. (The opcode is included in the message.) The server
will return an error code of NOTIMPL to the sender.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_XFRIN_CHANNEL_CREATED">
<term>AUTH_XFRIN_CHANNEL_CREATED XFRIN session channel created</term>
<listitem><para>
This is a debug message indicating that the authoritative server has
created a channel to the XFRIN (Transfer-in) process.  It is issued
during server startup is an indication that the initialization is
proceeding normally.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_XFRIN_CHANNEL_ESTABLISHED">
<term>AUTH_XFRIN_CHANNEL_ESTABLISHED XFRIN session channel established</term>
<listitem><para>
This is a debug message indicating that the authoritative server has
established communication over the previously-created channel to the
XFRIN (Transfer-in) process.  It is issued during server startup is an
indication that the initialization is proceeding normally.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_ZONEMGR_COMMS">
<term>AUTH_ZONEMGR_COMMS error communicating with zone manager: %1</term>
<listitem><para>
This is a debug message output during the processing of a NOTIFY request.
An error (listed in the message) has been encountered whilst communicating
with the zone manager. The NOTIFY request will not be honored.
</para></listitem>
</varlistentry>

<varlistentry id="AUTH_ZONEMGR_ERROR">
<term>AUTH_ZONEMGR_ERROR received error response from zone manager: %1</term>
<listitem><para>
This is a debug message output during the processing of a NOTIFY
request. The zone manager component has been informed of the request,
but has returned an error response (which is included in the message). The
NOTIFY request will not be honored.
</para></listitem>
</varlistentry>

<varlistentry id="CC_ASYNC_READ_FAILED">
<term>CC_ASYNC_READ_FAILED asynchronous read failed</term>
<listitem><para>
This marks a low level error, we tried to read data from the message queue
daemon asynchronously, but the ASIO library returned an error.
</para></listitem>
</varlistentry>

<varlistentry id="CC_CONN_ERROR">
<term>CC_CONN_ERROR error connecting to message queue (%1)</term>
<listitem><para>
It is impossible to reach the message queue daemon for the reason given. It
is unlikely there'll be reason for whatever program this currently is to
continue running, as the communication with the rest of BIND 10 is vital
for the components.
</para></listitem>
</varlistentry>

<varlistentry id="CC_DISCONNECT">
<term>CC_DISCONNECT disconnecting from message queue daemon</term>
<listitem><para>
The library is disconnecting from the message queue daemon. This debug message
indicates that the program is trying to shut down gracefully.
</para></listitem>
</varlistentry>

<varlistentry id="CC_ESTABLISH">
<term>CC_ESTABLISH trying to establish connection with message queue daemon at %1</term>
<listitem><para>
This debug message indicates that the command channel library is about to
connect to the message queue daemon, which should be listening on the UNIX-domain
socket listed in the output.
</para></listitem>
</varlistentry>

<varlistentry id="CC_ESTABLISHED">
<term>CC_ESTABLISHED successfully connected to message queue daemon</term>
<listitem><para>
This debug message indicates that the connection was successfully made, this
should follow CC_ESTABLISH.
</para></listitem>
</varlistentry>

<varlistentry id="CC_GROUP_RECEIVE">
<term>CC_GROUP_RECEIVE trying to receive a message</term>
<listitem><para>
Debug message, noting that a message is expected to come over the command
channel.
</para></listitem>
</varlistentry>

<varlistentry id="CC_GROUP_RECEIVED">
<term>CC_GROUP_RECEIVED message arrived ('%1', '%2')</term>
<listitem><para>
Debug message, noting that we successfully received a message (its envelope and
payload listed). This follows CC_GROUP_RECEIVE, but might happen some time
later, depending if we waited for it or just polled.
</para></listitem>
</varlistentry>

<varlistentry id="CC_GROUP_SEND">
<term>CC_GROUP_SEND sending message '%1' to group '%2'</term>
<listitem><para>
Debug message, we're about to send a message over the command channel.
</para></listitem>
</varlistentry>

<varlistentry id="CC_INVALID_LENGTHS">
<term>CC_INVALID_LENGTHS invalid length parameters (%1, %2)</term>
<listitem><para>
This happens when garbage comes over the command channel or some kind of
confusion happens in the program. The data received from the socket make no
sense if we interpret it as lengths of message. The first one is total length
of message, the second length of the header. The header and it's length
(2 bytes) is counted in the total length.
</para></listitem>
</varlistentry>

<varlistentry id="CC_LENGTH_NOT_READY">
<term>CC_LENGTH_NOT_READY length not ready</term>
<listitem><para>
There should be data representing length of message on the socket, but it
is not there.
</para></listitem>
</varlistentry>

<varlistentry id="CC_NO_MESSAGE">
<term>CC_NO_MESSAGE no message ready to be received yet</term>
<listitem><para>
The program polled for incoming messages, but there was no message waiting.
This is a debug message which may happen only after CC_GROUP_RECEIVE.
</para></listitem>
</varlistentry>

<varlistentry id="CC_NO_MSGQ">
<term>CC_NO_MSGQ unable to connect to message queue (%1)</term>
<listitem><para>
It isn't possible to connect to the message queue daemon, for reason listed.
It is unlikely any program will be able continue without the communication.
</para></listitem>
</varlistentry>

<varlistentry id="CC_READ_ERROR">
<term>CC_READ_ERROR error reading data from command channel (%1)</term>
<listitem><para>
A low level error happened when the library tried to read data from the
command channel socket. The reason is listed.
</para></listitem>
</varlistentry>

<varlistentry id="CC_READ_EXCEPTION">
<term>CC_READ_EXCEPTION error reading data from command channel (%1)</term>
<listitem><para>
We received an exception while trying to read data from the command
channel socket. The reason is listed.
</para></listitem>
</varlistentry>

<varlistentry id="CC_REPLY">
<term>CC_REPLY replying to message from '%1' with '%2'</term>
<listitem><para>
Debug message, noting we're sending a response to the original message
with the given envelope.
</para></listitem>
</varlistentry>

<varlistentry id="CC_SET_TIMEOUT">
<term>CC_SET_TIMEOUT setting timeout to %1ms</term>
<listitem><para>
Debug message. A timeout for which the program is willing to wait for a reply
is being set.
</para></listitem>
</varlistentry>

<varlistentry id="CC_START_READ">
<term>CC_START_READ starting asynchronous read</term>
<listitem><para>
Debug message. From now on, when a message (or command) comes, it'll wake the
program and the library will automatically pass it over to correct place.
</para></listitem>
</varlistentry>

<varlistentry id="CC_SUBSCRIBE">
<term>CC_SUBSCRIBE subscribing to communication group %1</term>
<listitem><para>
Debug message. The program wants to receive messages addressed to this group.
</para></listitem>
</varlistentry>

<varlistentry id="CC_TIMEOUT">
<term>CC_TIMEOUT timeout reading data from command channel</term>
<listitem><para>
The program waited too long for data from the command channel (usually when it
sent a query to different program and it didn't answer for whatever reason).
</para></listitem>
</varlistentry>

<varlistentry id="CC_UNSUBSCRIBE">
<term>CC_UNSUBSCRIBE unsubscribing from communication group %1</term>
<listitem><para>
Debug message. The program no longer wants to receive messages addressed to
this group.
</para></listitem>
</varlistentry>

<varlistentry id="CC_WRITE_ERROR">
<term>CC_WRITE_ERROR error writing data to command channel (%1)</term>
<listitem><para>
A low level error happened when the library tried to write data to the command
channel socket.
</para></listitem>
</varlistentry>

<varlistentry id="CC_ZERO_LENGTH">
<term>CC_ZERO_LENGTH invalid message length (0)</term>
<listitem><para>
The library received a message length being zero, which makes no sense, since
all messages must contain at least the envelope.
</para></listitem>
</varlistentry>

<varlistentry id="CFGMGR_AUTOMATIC_CONFIG_DATABASE_UPDATE">
<term>CFGMGR_AUTOMATIC_CONFIG_DATABASE_UPDATE Updating configuration database from version %1 to %2</term>
<listitem><para>
An older version of the configuration database has been found, from which
there was an automatic upgrade path to the current version. These changes
are now applied, and no action from the administrator is necessary.
</para></listitem>
</varlistentry>

<varlistentry id="CFGMGR_CC_SESSION_ERROR">
<term>CFGMGR_CC_SESSION_ERROR Error connecting to command channel: %1</term>
<listitem><para>
The configuration manager daemon was unable to connect to the messaging
system. The most likely cause is that msgq is not running.
</para></listitem>
</varlistentry>

<varlistentry id="CFGMGR_DATA_READ_ERROR">
<term>CFGMGR_DATA_READ_ERROR error reading configuration database from disk: %1</term>
<listitem><para>
There was a problem reading the persistent configuration data as stored
on disk. The file may be corrupted, or it is of a version from where
there is no automatic upgrade path. The file needs to be repaired or
removed. The configuration manager daemon will now shut down.
</para></listitem>
</varlistentry>

<varlistentry id="CFGMGR_IOERROR_WHILE_WRITING_CONFIGURATION">
<term>CFGMGR_IOERROR_WHILE_WRITING_CONFIGURATION Unable to write configuration file; configuration not stored: %1</term>
<listitem><para>
There was an IO error from the system while the configuration manager
was trying to write the configuration database to disk. The specific
error is given. The most likely cause is that the directory where
the file is stored does not exist, or is not writable. The updated
configuration is not stored.
</para></listitem>
</varlistentry>

<varlistentry id="CFGMGR_OSERROR_WHILE_WRITING_CONFIGURATION">
<term>CFGMGR_OSERROR_WHILE_WRITING_CONFIGURATION Unable to write configuration file; configuration not stored: %1</term>
<listitem><para>
There was an OS error from the system while the configuration manager
was trying to write the configuration database to disk. The specific
error is given. The most likely cause is that the system does not have
write access to the configuration database file. The updated
configuration is not stored.
</para></listitem>
</varlistentry>

<varlistentry id="CFGMGR_STOPPED_BY_KEYBOARD">
<term>CFGMGR_STOPPED_BY_KEYBOARD keyboard interrupt, shutting down</term>
<listitem><para>
There was a keyboard interrupt signal to stop the cfgmgr daemon. The
daemon will now shut down.
</para></listitem>
</varlistentry>

<varlistentry id="CONFIG_CCSESSION_MSG">
<term>CONFIG_CCSESSION_MSG error in CC session message: %1</term>
<listitem><para>
There was a problem with an incoming message on the command and control
channel. The message does not appear to be a valid command, and is
missing a required element or contains an unknown data format. This
most likely means that another BIND10 module is sending a bad message.
The message itself is ignored by this module.
</para></listitem>
</varlistentry>

<varlistentry id="CONFIG_CCSESSION_MSG_INTERNAL">
<term>CONFIG_CCSESSION_MSG_INTERNAL error handling CC session message: %1</term>
<listitem><para>
There was an internal problem handling an incoming message on the command
and control channel. An unexpected exception was thrown, details of
which are appended to the message. The module will continue to run,
but will not send back an answer.
</para><para>
The most likely cause of this error is a programming error.  Please raise
a bug report.
</para></listitem>
</varlistentry>

<varlistentry id="CONFIG_GET_FAIL">
<term>CONFIG_GET_FAIL error getting configuration from cfgmgr: %1</term>
<listitem><para>
The configuration manager returned an error when this module requested
the configuration. The full error message answer from the configuration
manager is appended to the log error. The most likely cause is that
the module is of a different (command specification) version than the
running configuration manager.
</para></listitem>
</varlistentry>

<varlistentry id="CONFIG_JSON_PARSE">
<term>CONFIG_JSON_PARSE JSON parse error in %1: %2</term>
<listitem><para>
There was an error parsing the JSON file. The given file does not appear
to be in valid JSON format. Please verify that the filename is correct
and that the contents are valid JSON.
</para></listitem>
</varlistentry>

<varlistentry id="CONFIG_MOD_SPEC_FORMAT">
<term>CONFIG_MOD_SPEC_FORMAT module specification error in %1: %2</term>
<listitem><para>
The given file does not appear to be a valid specification file: details
are included in the message. Please verify that the filename is correct
and that its contents are a valid BIND10 module specification.
</para></listitem>
</varlistentry>

<varlistentry id="CONFIG_MOD_SPEC_REJECT">
<term>CONFIG_MOD_SPEC_REJECT module specification rejected by cfgmgr: %1</term>
<listitem><para>
The specification file for this module was rejected by the configuration
manager. The full error message answer from the configuration manager is
appended to the log error. The most likely cause is that the module is of
a different (specification file) version than the running configuration
manager.
</para></listitem>
</varlistentry>

<varlistentry id="CONFIG_OPEN_FAIL">
<term>CONFIG_OPEN_FAIL error opening %1: %2</term>
<listitem><para>
There was an error opening the given file. The reason for the failure
is included in the message.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_CREATE">
<term>DATASRC_CACHE_CREATE creating the hotspot cache</term>
<listitem><para>
Debug information that the hotspot cache was created at startup.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_DESTROY">
<term>DATASRC_CACHE_DESTROY destroying the hotspot cache</term>
<listitem><para>
Debug information. The hotspot cache is being destroyed.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_DISABLE">
<term>DATASRC_CACHE_DISABLE disabling the cache</term>
<listitem><para>
The hotspot cache is disabled from now on. It is not going to store
information or return anything.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_ENABLE">
<term>DATASRC_CACHE_ENABLE enabling the cache</term>
<listitem><para>
The hotspot cache is enabled from now on.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_EXPIRED">
<term>DATASRC_CACHE_EXPIRED the item '%1' is expired</term>
<listitem><para>
Debug information. There was an attempt to look up an item in the hotspot
cache. And the item was actually there, but it was too old, so it was removed
instead and nothing is reported (the external behaviour is the same as with
CACHE_NOT_FOUND).
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_FOUND">
<term>DATASRC_CACHE_FOUND the item '%1' was found</term>
<listitem><para>
Debug information. An item was successfully looked up in the hotspot cache.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_FULL">
<term>DATASRC_CACHE_FULL cache is full, dropping oldest</term>
<listitem><para>
Debug information. After inserting an item into the hotspot cache, the
maximum number of items was exceeded, so the least recently used item will
be dropped. This should be directly followed by CACHE_REMOVE.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_INSERT">
<term>DATASRC_CACHE_INSERT inserting item '%1' into the cache</term>
<listitem><para>
Debug information. It means a new item is being inserted into the hotspot
cache.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_NOT_FOUND">
<term>DATASRC_CACHE_NOT_FOUND the item '%1' was not found</term>
<listitem><para>
Debug information. It was attempted to look up an item in the hotspot cache,
but it is not there.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_OLD_FOUND">
<term>DATASRC_CACHE_OLD_FOUND older instance of cache item found, replacing</term>
<listitem><para>
Debug information. While inserting an item into the hotspot cache, an older
instance of an item with the same name was found. The old instance will be
removed. This should be directly followed by CACHE_REMOVE.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_REMOVE">
<term>DATASRC_CACHE_REMOVE removing '%1' from the cache</term>
<listitem><para>
Debug information. An item is being removed from the hotspot cache.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_CACHE_SLOTS">
<term>DATASRC_CACHE_SLOTS setting the cache size to '%1', dropping '%2' items</term>
<listitem><para>
The maximum allowed number of items of the hotspot cache is set to the given
number. If there are too many, some of them will be dropped. The size of 0
means no limit.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_DO_QUERY">
<term>DATASRC_DO_QUERY handling query for '%1/%2'</term>
<listitem><para>
Debug information. We're processing some internal query for given name and
type.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_ADD_RRSET">
<term>DATASRC_MEM_ADD_RRSET adding RRset '%1/%2' into zone '%3'</term>
<listitem><para>
Debug information. An RRset is being added to the in-memory data source.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_ADD_WILDCARD">
<term>DATASRC_MEM_ADD_WILDCARD adding wildcards for '%1'</term>
<listitem><para>
Debug information. Some special marks above each * in wildcard name are needed.
They are being added now for this name.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_ADD_ZONE">
<term>DATASRC_MEM_ADD_ZONE adding zone '%1/%2'</term>
<listitem><para>
Debug information. A zone is being added into the in-memory data source.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_ANY_SUCCESS">
<term>DATASRC_MEM_ANY_SUCCESS ANY query for '%1' successful</term>
<listitem><para>
Debug information. The domain was found and an ANY type query is being answered
by providing everything found inside the domain.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_CNAME">
<term>DATASRC_MEM_CNAME CNAME at the domain '%1'</term>
<listitem><para>
Debug information. The requested domain is an alias to a different domain,
returning the CNAME instead.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_CNAME_COEXIST">
<term>DATASRC_MEM_CNAME_COEXIST can't add data to CNAME in domain '%1'</term>
<listitem><para>
This is the same problem as in MEM_CNAME_TO_NONEMPTY, but it happened the
other way around -- adding some other data to CNAME.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_CNAME_TO_NONEMPTY">
<term>DATASRC_MEM_CNAME_TO_NONEMPTY can't add CNAME to domain with other data in '%1'</term>
<listitem><para>
Someone or something tried to add a CNAME into a domain that already contains
some other data. But the protocol forbids coexistence of CNAME with anything
(RFC 1034, section 3.6.2). This indicates a problem with provided data.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_CREATE">
<term>DATASRC_MEM_CREATE creating zone '%1' in '%2' class</term>
<listitem><para>
Debug information. A representation of a zone for the in-memory data source is
being created.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_DELEG_FOUND">
<term>DATASRC_MEM_DELEG_FOUND delegation found at '%1'</term>
<listitem><para>
Debug information. A delegation point was found above the requested record.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_DESTROY">
<term>DATASRC_MEM_DESTROY destroying zone '%1' in '%2' class</term>
<listitem><para>
Debug information. A zone from in-memory data source is being destroyed.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_DNAME_ENCOUNTERED">
<term>DATASRC_MEM_DNAME_ENCOUNTERED encountered a DNAME</term>
<listitem><para>
Debug information. While searching for the requested domain, a DNAME was
encountered on the way.  This may lead to redirection to a different domain and
stop the search.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_DNAME_FOUND">
<term>DATASRC_MEM_DNAME_FOUND DNAME found at '%1'</term>
<listitem><para>
Debug information. A DNAME was found instead of the requested information.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_DNAME_NS">
<term>DATASRC_MEM_DNAME_NS DNAME and NS can't coexist in non-apex domain '%1'</term>
<listitem><para>
It was requested for DNAME and NS records to be put into the same domain
which is not the apex (the top of the zone). This is forbidden by RFC
2672, section 3. This indicates a problem with provided data.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_DOMAIN_EMPTY">
<term>DATASRC_MEM_DOMAIN_EMPTY requested domain '%1' is empty</term>
<listitem><para>
Debug information. The requested domain exists in the tree of domains, but
it is empty. Therefore it doesn't contain the requested resource type.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_DUP_RRSET">
<term>DATASRC_MEM_DUP_RRSET duplicate RRset '%1/%2'</term>
<listitem><para>
An RRset is being inserted into in-memory data source for a second time.  The
original version must be removed first. Note that loading master files where an
RRset is split into multiple locations is not supported yet.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_EXACT_DELEGATION">
<term>DATASRC_MEM_EXACT_DELEGATION delegation at the exact domain '%1'</term>
<listitem><para>
Debug information. There's a NS record at the requested domain. This means
this zone is not authoritative for the requested domain, but a delegation
should be followed. The requested domain is an apex of some zone.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_FIND">
<term>DATASRC_MEM_FIND find '%1/%2'</term>
<listitem><para>
Debug information. A search for the requested RRset is being started.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_FIND_ZONE">
<term>DATASRC_MEM_FIND_ZONE looking for zone '%1'</term>
<listitem><para>
Debug information. A zone object for this zone is being searched for in the
in-memory data source.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_LOAD">
<term>DATASRC_MEM_LOAD loading zone '%1' from file '%2'</term>
<listitem><para>
Debug information. The content of master file is being loaded into the memory.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_NOTFOUND">
<term>DATASRC_MEM_NOTFOUND requested domain '%1' not found</term>
<listitem><para>
Debug information. The requested domain does not exist.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_NS_ENCOUNTERED">
<term>DATASRC_MEM_NS_ENCOUNTERED encountered a NS</term>
<listitem><para>
Debug information. While searching for the requested domain, a NS was
encountered on the way (a delegation). This may lead to stop of the search.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_NXRRSET">
<term>DATASRC_MEM_NXRRSET no such type '%1' at '%2'</term>
<listitem><para>
Debug information. The domain exists, but it doesn't hold any record of the
requested type.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_OUT_OF_ZONE">
<term>DATASRC_MEM_OUT_OF_ZONE domain '%1' doesn't belong to zone '%2'</term>
<listitem><para>
It was attempted to add the domain into a zone that shouldn't have it
(eg. the domain is not subdomain of the zone origin). This indicates a
problem with provided data.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_RENAME">
<term>DATASRC_MEM_RENAME renaming RRset from '%1' to '%2'</term>
<listitem><para>
Debug information. A RRset is being generated from a different RRset (most
probably a wildcard). So it must be renamed to whatever the user asked for. In
fact, it's impossible to rename RRsets with our libraries, so a new one is
created and all resource records are copied over.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_SINGLETON">
<term>DATASRC_MEM_SINGLETON trying to add multiple RRs for domain '%1' and type '%2'</term>
<listitem><para>
Some resource types are singletons -- only one is allowed in a domain
(for example CNAME or SOA). This indicates a problem with provided data.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_SUCCESS">
<term>DATASRC_MEM_SUCCESS query for '%1/%2' successful</term>
<listitem><para>
Debug information. The requested record was found.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_SUPER_STOP">
<term>DATASRC_MEM_SUPER_STOP stopped at superdomain '%1', domain '%2' is empty</term>
<listitem><para>
Debug information. The search stopped at a superdomain of the requested
domain. The domain is a empty nonterminal, therefore it is treated  as NXRRSET
case (eg. the domain exists, but it doesn't have the requested record type).
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_SWAP">
<term>DATASRC_MEM_SWAP swapping contents of two zone representations ('%1' and '%2')</term>
<listitem><para>
Debug information. The contents of two in-memory zones are being exchanged.
This is usual practice to do some manipulation in exception-safe manner -- the
new data are prepared in a different zone object and when it works, they are
swapped. The old one contains the new data and the other one can be safely
destroyed.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_WILDCARD_CANCEL">
<term>DATASRC_MEM_WILDCARD_CANCEL wildcard match canceled for '%1'</term>
<listitem><para>
Debug information. A domain above wildcard was reached, but there's something
below the requested domain. Therefore the wildcard doesn't apply here.  This
behaviour is specified by RFC 1034, section 4.3.3
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_WILDCARD_DNAME">
<term>DATASRC_MEM_WILDCARD_DNAME DNAME record in wildcard domain '%1'</term>
<listitem><para>
The software refuses to load DNAME records into a wildcard domain.  It isn't
explicitly forbidden, but the protocol is ambiguous about how this should
behave and BIND 9 refuses that as well. Please describe your intention using
different tools.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_MEM_WILDCARD_NS">
<term>DATASRC_MEM_WILDCARD_NS NS record in wildcard domain '%1'</term>
<listitem><para>
The software refuses to load NS records into a wildcard domain.  It isn't
explicitly forbidden, but the protocol is ambiguous about how this should
behave and BIND 9 refuses that as well. Please describe your intention using
different tools.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_META_ADD">
<term>DATASRC_META_ADD adding a data source into meta data source</term>
<listitem><para>
Debug information. Yet another data source is being added into the meta data
source. (probably at startup or reconfiguration)
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_META_ADD_CLASS_MISMATCH">
<term>DATASRC_META_ADD_CLASS_MISMATCH mismatch between classes '%1' and '%2'</term>
<listitem><para>
It was attempted to add a data source into a meta data source. But their
classes do not match.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_META_REMOVE">
<term>DATASRC_META_REMOVE removing data source from meta data source</term>
<listitem><para>
Debug information. A data source is being removed from meta data source.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_ADD_NSEC">
<term>DATASRC_QUERY_ADD_NSEC adding NSEC record for '%1'</term>
<listitem><para>
Debug information. A NSEC record covering this zone is being added.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_ADD_NSEC3">
<term>DATASRC_QUERY_ADD_NSEC3 adding NSEC3 record of zone '%1'</term>
<listitem><para>
Debug information. A NSEC3 record for the given zone is being added to the
response message.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_ADD_RRSET">
<term>DATASRC_QUERY_ADD_RRSET adding RRset '%1/%2' to message</term>
<listitem><para>
Debug information. An RRset is being added to the response message.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_ADD_SOA">
<term>DATASRC_QUERY_ADD_SOA adding SOA of '%1'</term>
<listitem><para>
Debug information. A SOA record of the given zone is being added to the
authority section of the response message.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_AUTH_FAIL">
<term>DATASRC_QUERY_AUTH_FAIL the underlying data source failed with %1</term>
<listitem><para>
The underlying data source failed to answer the authoritative query. 1 means
some error, 2 is not implemented. The data source should have logged the
specific error already.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_BAD_REFERRAL">
<term>DATASRC_QUERY_BAD_REFERRAL bad referral to '%1'</term>
<listitem><para>
The domain lives in another zone. But it is not possible to generate referral
information for it.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_CACHED">
<term>DATASRC_QUERY_CACHED data for %1/%2 found in cache</term>
<listitem><para>
Debug information. The requested data were found in the hotspot cache, so
no query is sent to the real data source.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_CHECK_CACHE">
<term>DATASRC_QUERY_CHECK_CACHE checking cache for '%1/%2'</term>
<listitem><para>
Debug information. While processing a query, lookup to the hotspot cache
is being made.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_COPY_AUTH">
<term>DATASRC_QUERY_COPY_AUTH copying authoritative section into message</term>
<listitem><para>
Debug information. The whole referral information is being copied into the
response message.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_DELEGATION">
<term>DATASRC_QUERY_DELEGATION looking for delegation on the path to '%1'</term>
<listitem><para>
Debug information. The software is trying to identify delegation points on the
way down to the given domain.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_EMPTY_CNAME">
<term>DATASRC_QUERY_EMPTY_CNAME CNAME at '%1' is empty</term>
<listitem><para>
There was an CNAME and it was being followed. But it contains no records,
so there's nowhere to go. There will be no answer. This indicates a problem
with supplied data.
We tried to follow
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_EMPTY_DNAME">
<term>DATASRC_QUERY_EMPTY_DNAME the DNAME on '%1' is empty</term>
<listitem><para>
During an attempt to synthesize CNAME from this DNAME it was discovered the
DNAME is empty (it has no records). This indicates problem with supplied data.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_FAIL">
<term>DATASRC_QUERY_FAIL query failed</term>
<listitem><para>
Some subtask of query processing failed. The reason should have been reported
already. We are returning SERVFAIL.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_FOLLOW_CNAME">
<term>DATASRC_QUERY_FOLLOW_CNAME following CNAME at '%1'</term>
<listitem><para>
Debug information. The domain is a CNAME (or a DNAME and we created a CNAME
for it already), so it's being followed.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_GET_MX_ADDITIONAL">
<term>DATASRC_QUERY_GET_MX_ADDITIONAL addition of A/AAAA for '%1' requested by MX '%2'</term>
<listitem><para>
Debug information. While processing a query, a MX record was met. It
references the mentioned address, so A/AAAA records for it are looked up
and put it into the additional section.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_GET_NS_ADDITIONAL">
<term>DATASRC_QUERY_GET_NS_ADDITIONAL addition of A/AAAA for '%1' requested by NS '%2'</term>
<listitem><para>
Debug information. While processing a query, a NS record was met. It
references the mentioned address, so A/AAAA records for it are looked up
and put it into the additional section.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_GLUE_FAIL">
<term>DATASRC_QUERY_GLUE_FAIL the underlying data source failed with %1</term>
<listitem><para>
The underlying data source failed to answer the glue query. 1 means some error,
2 is not implemented. The data source should have logged the specific error
already.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_INVALID_OP">
<term>DATASRC_QUERY_INVALID_OP invalid query operation requested</term>
<listitem><para>
This indicates a programmer error. The DO_QUERY was called with unknown
operation code.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_IS_AUTH">
<term>DATASRC_QUERY_IS_AUTH auth query (%1/%2)</term>
<listitem><para>
Debug information. The last DO_QUERY is an auth query.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_IS_GLUE">
<term>DATASRC_QUERY_IS_GLUE glue query (%1/%2)</term>
<listitem><para>
Debug information. The last DO_QUERY is query for glue addresses.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_IS_NOGLUE">
<term>DATASRC_QUERY_IS_NOGLUE query for non-glue addresses (%1/%2)</term>
<listitem><para>
Debug information. The last DO_QUERY is query for addresses that are not
glue.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_IS_REF">
<term>DATASRC_QUERY_IS_REF query for referral (%1/%2)</term>
<listitem><para>
Debug information. The last DO_QUERY is query for referral information.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_IS_SIMPLE">
<term>DATASRC_QUERY_IS_SIMPLE simple query (%1/%2)</term>
<listitem><para>
Debug information. The last DO_QUERY is a simple query.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_MISPLACED_TASK">
<term>DATASRC_QUERY_MISPLACED_TASK task of this type should not be here</term>
<listitem><para>
This indicates a programming error. A task was found in the internal task
queue, but this kind of task wasn't designed to be inside the queue (it should
be handled right away, not queued).
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_MISSING_NS">
<term>DATASRC_QUERY_MISSING_NS missing NS records for '%1'</term>
<listitem><para>
NS records should have been put into the authority section. However, this zone
has none. This indicates problem with provided data.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_MISSING_SOA">
<term>DATASRC_QUERY_MISSING_SOA the zone '%1' has no SOA</term>
<listitem><para>
The answer should have been a negative one (eg. of nonexistence of something).
To do so, a SOA record should be put into the authority section, but the zone
does not have one. This indicates problem with provided data.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_NOGLUE_FAIL">
<term>DATASRC_QUERY_NOGLUE_FAIL the underlying data source failed with %1</term>
<listitem><para>
The underlying data source failed to answer the no-glue query. 1 means some
error, 2 is not implemented. The data source should have logged the specific
error already.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_NO_CACHE_ANY_AUTH">
<term>DATASRC_QUERY_NO_CACHE_ANY_AUTH ignoring cache for ANY query (%1/%2 in %3 class)</term>
<listitem><para>
Debug information. The hotspot cache is ignored for authoritative ANY queries
for consistency reasons.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_NO_CACHE_ANY_SIMPLE">
<term>DATASRC_QUERY_NO_CACHE_ANY_SIMPLE ignoring cache for ANY query (%1/%2 in %3 class)</term>
<listitem><para>
Debug information. The hotspot cache is ignored for ANY queries for consistency
reasons.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_NO_DS_NSEC">
<term>DATASRC_QUERY_NO_DS_NSEC there's no DS record in the '%1' zone</term>
<listitem><para>
An attempt to add a NSEC record into the message failed, because the zone does
not have any DS record. This indicates problem with the provided data.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_NO_DS_NSEC3">
<term>DATASRC_QUERY_NO_DS_NSEC3 there's no DS record in the '%1' zone</term>
<listitem><para>
An attempt to add a NSEC3 record into the message failed, because the zone does
not have any DS record. This indicates problem with the provided data.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_NO_ZONE">
<term>DATASRC_QUERY_NO_ZONE no zone containing '%1' in class '%2'</term>
<listitem><para>
Lookup of domain failed because the data have no zone that contain the
domain. Maybe someone sent a query to the wrong server for some reason.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_PROCESS">
<term>DATASRC_QUERY_PROCESS processing query '%1/%2' in the '%3' class</term>
<listitem><para>
Debug information. A sure query is being processed now.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_PROVENX_FAIL">
<term>DATASRC_QUERY_PROVENX_FAIL unable to prove nonexistence of '%1'</term>
<listitem><para>
The user wants DNSSEC and we discovered the entity doesn't exist (either
domain or the record). But there was an error getting NSEC/NSEC3 record
to prove the nonexistence.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_REF_FAIL">
<term>DATASRC_QUERY_REF_FAIL the underlying data source failed with %1</term>
<listitem><para>
The underlying data source failed to answer the query for referral information.
1 means some error, 2 is not implemented. The data source should have logged
the specific error already.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_RRSIG">
<term>DATASRC_QUERY_RRSIG unable to answer RRSIG query</term>
<listitem><para>
The server is unable to answer a direct query for RRSIG type, but was asked
to do so.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_SIMPLE_FAIL">
<term>DATASRC_QUERY_SIMPLE_FAIL the underlying data source failed with %1</term>
<listitem><para>
The underlying data source failed to answer the simple query. 1 means some
error, 2 is not implemented. The data source should have logged the specific
error already.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_SYNTH_CNAME">
<term>DATASRC_QUERY_SYNTH_CNAME synthesizing CNAME from DNAME on '%1'</term>
<listitem><para>
Debug information. While answering a query, a DNAME was met. The DNAME itself
will be returned, but along with it a CNAME for clients which don't understand
DNAMEs will be synthesized.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_TASK_FAIL">
<term>DATASRC_QUERY_TASK_FAIL task failed with %1</term>
<listitem><para>
The query subtask failed. The reason should have been reported by the subtask
already. The code is 1 for error, 2 for not implemented.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_TOO_MANY_CNAMES">
<term>DATASRC_QUERY_TOO_MANY_CNAMES CNAME chain limit exceeded at '%1'</term>
<listitem><para>
A CNAME led to another CNAME and it led to another, and so on. After 16
CNAMEs, the software gave up. Long CNAME chains are discouraged, and this
might possibly be a loop as well. Note that some of the CNAMEs might have
been synthesized from DNAMEs. This indicates problem with supplied data.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_UNKNOWN_RESULT">
<term>DATASRC_QUERY_UNKNOWN_RESULT unknown result of subtask</term>
<listitem><para>
This indicates a programmer error. The answer of subtask doesn't look like
anything known.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_WILDCARD">
<term>DATASRC_QUERY_WILDCARD looking for a wildcard covering '%1'</term>
<listitem><para>
Debug information. A direct match wasn't found, so a wildcard covering the
domain is being looked for now.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_WILDCARD_FAIL">
<term>DATASRC_QUERY_WILDCARD_FAIL error processing wildcard for '%1'</term>
<listitem><para>
During an attempt to cover the domain by a wildcard an error happened. The
exact kind was hopefully already reported.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_WILDCARD_PROVENX_FAIL">
<term>DATASRC_QUERY_WILDCARD_PROVENX_FAIL unable to prove nonexistence of '%1' (%2)</term>
<listitem><para>
While processing a wildcard, it wasn't possible to prove nonexistence of the
given domain or record.  The code is 1 for error and 2 for not implemented.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_QUERY_WILDCARD_REFERRAL">
<term>DATASRC_QUERY_WILDCARD_REFERRAL unable to find referral info for '%1' (%2)</term>
<listitem><para>
While processing a wildcard, a referral was met. But it wasn't possible to get
enough information for it.  The code is 1 for error, 2 for not implemented.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_CLOSE">
<term>DATASRC_SQLITE_CLOSE closing SQLite database</term>
<listitem><para>
Debug information. The SQLite data source is closing the database file.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_CREATE">
<term>DATASRC_SQLITE_CREATE SQLite data source created</term>
<listitem><para>
Debug information. An instance of SQLite data source is being created.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_DESTROY">
<term>DATASRC_SQLITE_DESTROY SQLite data source destroyed</term>
<listitem><para>
Debug information. An instance of SQLite data source is being destroyed.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_ENCLOSURE">
<term>DATASRC_SQLITE_ENCLOSURE looking for zone containing '%1'</term>
<listitem><para>
Debug information. The SQLite data source is trying to identify which zone
should hold this domain.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_ENCLOSURE_NOTFOUND">
<term>DATASRC_SQLITE_ENCLOSURE_NOTFOUND no zone contains it</term>
<listitem><para>
Debug information. The last SQLITE_ENCLOSURE query was unsuccessful; there's
no such zone in our data.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_FIND">
<term>DATASRC_SQLITE_FIND looking for RRset '%1/%2'</term>
<listitem><para>
Debug information. The SQLite data source is looking up a resource record
set.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_FINDADDRS">
<term>DATASRC_SQLITE_FINDADDRS looking for A/AAAA addresses for '%1'</term>
<listitem><para>
Debug information. The data source is looking up the addresses for given
domain name.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_FINDADDRS_BAD_CLASS">
<term>DATASRC_SQLITE_FINDADDRS_BAD_CLASS class mismatch looking for addresses ('%1' and '%2')</term>
<listitem><para>
The SQLite data source was looking up A/AAAA addresses, but the data source
contains different class than the query was for.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_FINDEXACT">
<term>DATASRC_SQLITE_FINDEXACT looking for exact RRset '%1/%2'</term>
<listitem><para>
Debug information. The SQLite data source is looking up an exact resource
record.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_FINDEXACT_BAD_CLASS">
<term>DATASRC_SQLITE_FINDEXACT_BAD_CLASS class mismatch looking for an RRset ('%1' and '%2')</term>
<listitem><para>
The SQLite data source was looking up an exact RRset, but the data source
contains different class than the query was for.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_FINDREC">
<term>DATASRC_SQLITE_FINDREC looking for record '%1/%2'</term>
<listitem><para>
Debug information. The SQLite data source is looking up records of given name
and type in the database.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_FINDREF">
<term>DATASRC_SQLITE_FINDREF looking for referral at '%1'</term>
<listitem><para>
Debug information. The SQLite data source is identifying if this domain is
a referral and where it goes.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_FINDREF_BAD_CLASS">
<term>DATASRC_SQLITE_FINDREF_BAD_CLASS class mismatch looking for referral ('%1' and '%2')</term>
<listitem><para>
The SQLite data source was trying to identify if there's a referral. But
it contains different class than the query was for.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_FIND_BAD_CLASS">
<term>DATASRC_SQLITE_FIND_BAD_CLASS class mismatch looking for an RRset ('%1' and '%2')</term>
<listitem><para>
The SQLite data source was looking up an RRset, but the data source contains
different class than the query was for.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_FIND_NSEC3">
<term>DATASRC_SQLITE_FIND_NSEC3 looking for NSEC3 in zone '%1' for hash '%2'</term>
<listitem><para>
Debug information. We're trying to look up a NSEC3 record in the SQLite data
source.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_FIND_NSEC3_NO_ZONE">
<term>DATASRC_SQLITE_FIND_NSEC3_NO_ZONE no such zone '%1'</term>
<listitem><para>
The SQLite data source was asked to provide a NSEC3 record for given zone.
But it doesn't contain that zone.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_OPEN">
<term>DATASRC_SQLITE_OPEN opening SQLite database '%1'</term>
<listitem><para>
Debug information. The SQLite data source is loading an SQLite database in
the provided file.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_PREVIOUS">
<term>DATASRC_SQLITE_PREVIOUS looking for name previous to '%1'</term>
<listitem><para>
Debug information. We're trying to look up name preceding the supplied one.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_PREVIOUS_NO_ZONE">
<term>DATASRC_SQLITE_PREVIOUS_NO_ZONE no zone containing '%1'</term>
<listitem><para>
The SQLite data source tried to identify name preceding this one. But this
one is not contained in any zone in the data source.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_SQLITE_SETUP">
<term>DATASRC_SQLITE_SETUP setting up SQLite database</term>
<listitem><para>
The database for SQLite data source was found empty. It is assumed this is the
first run and it is being initialized with current schema.  It'll still contain
no data, but it will be ready for use.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_STATIC_BAD_CLASS">
<term>DATASRC_STATIC_BAD_CLASS static data source can handle CH only</term>
<listitem><para>
For some reason, someone asked the static data source a query that is not in
the CH class.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_STATIC_CREATE">
<term>DATASRC_STATIC_CREATE creating the static datasource</term>
<listitem><para>
Debug information. The static data source (the one holding stuff like
version.bind) is being created.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_STATIC_FIND">
<term>DATASRC_STATIC_FIND looking for '%1/%2'</term>
<listitem><para>
Debug information. This resource record set is being looked up in the static
data source.
</para></listitem>
</varlistentry>

<varlistentry id="DATASRC_UNEXPECTED_QUERY_STATE">
<term>DATASRC_UNEXPECTED_QUERY_STATE unexpected query state</term>
<listitem><para>
This indicates a programming error. An internal task of unknown type was
generated.
</para></listitem>
</varlistentry>

<varlistentry id="LOGIMPL_ABOVE_MAX_DEBUG">
<term>LOGIMPL_ABOVE_MAX_DEBUG debug level of %1 is too high and will be set to the maximum of %2</term>
<listitem><para>
A message from the interface to the underlying logger implementation reporting
that the debug level (as set by an internally-created string DEBUGn, where n
is an integer, e.g. DEBUG22) is above the maximum allowed value and has
been reduced to that value.  The appearance of this message may indicate
a programming error - please submit a bug report.
</para></listitem>
</varlistentry>

<varlistentry id="LOGIMPL_BAD_DEBUG_STRING">
<term>LOGIMPL_BAD_DEBUG_STRING debug string '%1' has invalid format</term>
<listitem><para>
A message from the interface to the underlying logger implementation
reporting that an internally-created string used to set the debug level
is not of the correct format (it should be of the form DEBUGn, where n
is an integer, e.g. DEBUG22).  The appearance of this message indicates
a programming error - please submit a bug report.
</para></listitem>
</varlistentry>

<varlistentry id="LOGIMPL_BELOW_MIN_DEBUG">
<term>LOGIMPL_BELOW_MIN_DEBUG debug level of %1 is too low and will be set to the minimum of %2</term>
<listitem><para>
A message from the interface to the underlying logger implementation reporting
that the debug level (as set by an internally-created string DEBUGn, where n
is an integer, e.g. DEBUG22) is below the minimum allowed value and has
been increased to that value.  The appearance of this message may indicate
a programming error - please submit a bug report.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_BAD_DESTINATION">
<term>LOG_BAD_DESTINATION unrecognized log destination: %1</term>
<listitem><para>
A logger destination value was given that was not recognized. The
destination should be one of "console", "file", or "syslog".
</para></listitem>
</varlistentry>

<varlistentry id="LOG_BAD_SEVERITY">
<term>LOG_BAD_SEVERITY unrecognized log severity: %1</term>
<listitem><para>
A logger severity value was given that was not recognized. The severity
should be one of "DEBUG", "INFO", "WARN", "ERROR", or "FATAL".
</para></listitem>
</varlistentry>

<varlistentry id="LOG_BAD_STREAM">
<term>LOG_BAD_STREAM bad log console output stream: %1</term>
<listitem><para>
A log console output stream was given that was not recognized. The output
stream should be one of "stdout", or "stderr"
</para></listitem>
</varlistentry>

<varlistentry id="LOG_DUPLICATE_MESSAGE_ID">
<term>LOG_DUPLICATE_MESSAGE_ID duplicate message ID (%1) in compiled code</term>
<listitem><para>
During start-up, BIND10 detected that the given message identification had
been defined multiple times in the BIND10 code.
</para><para>
This has no ill-effects other than the possibility that an erronous
message may be logged.  However, as it is indicative of a programming
error, please log a bug report.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_DUPLICATE_NAMESPACE">
<term>LOG_DUPLICATE_NAMESPACE line %1: duplicate $NAMESPACE directive found</term>
<listitem><para>
When reading a message file, more than one $NAMESPACE directive was found.
Such a condition is regarded as an error and the read will be abandoned.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_INPUT_OPEN_FAIL">
<term>LOG_INPUT_OPEN_FAIL unable to open message file %1 for input: %2</term>
<listitem><para>
The program was not able to open the specified input message file for
the reason given.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_INVALID_MESSAGE_ID">
<term>LOG_INVALID_MESSAGE_ID line %1: invalid message identification '%2'</term>
<listitem><para>
An invalid message identification (ID) has been found during the read of
a message file.  Message IDs should comprise only alphanumeric characters
and the underscore, and should not start with a digit.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_NAMESPACE_EXTRA_ARGS">
<term>LOG_NAMESPACE_EXTRA_ARGS line %1: $NAMESPACE directive has too many arguments</term>
<listitem><para>
The $NAMESPACE directive in a message file takes a single argument, a
namespace in which all the generated symbol names are placed.  This error
is generated when the compiler finds a $NAMESPACE directive with more
than one argument.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_NAMESPACE_INVALID_ARG">
<term>LOG_NAMESPACE_INVALID_ARG line %1: $NAMESPACE directive has an invalid argument ('%2')</term>
<listitem><para>
The $NAMESPACE argument in a message file should be a valid C++ namespace.
This message is output if the simple check on the syntax of the string
carried out by the reader fails.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_NAMESPACE_NO_ARGS">
<term>LOG_NAMESPACE_NO_ARGS line %1: no arguments were given to the $NAMESPACE directive</term>
<listitem><para>
The $NAMESPACE directive in a message file takes a single argument,
a C++ namespace in which all the generated symbol names are placed.
This error is generated when the compiler finds a $NAMESPACE directive
with no arguments.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_NO_MESSAGE_ID">
<term>LOG_NO_MESSAGE_ID line %1: message definition line found without a message ID</term>
<listitem><para>
Within a message file, message are defined by lines starting with a "%".
The rest of the line should comprise the message ID and text describing
the message.  This error indicates the message compiler found a line in
the message file comprising just the "%" and nothing else.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_NO_MESSAGE_TEXT">
<term>LOG_NO_MESSAGE_TEXT line %1: line found containing a message ID ('%2') and no text</term>
<listitem><para>
Within a message file, message are defined by lines starting with a "%".
The rest of the line should comprise the message ID and text describing
the message.  This error indicates the message compiler found a line
in the message file comprising just the "%" and message identification,
but no text.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_NO_SUCH_MESSAGE">
<term>LOG_NO_SUCH_MESSAGE could not replace message text for '%1': no such message</term>
<listitem><para>
During start-up a local message file was read.  A line with the listed
message identification was found in the file, but the identification is
not one contained in the compiled-in message dictionary.  This message
may appear a number of times in the file, once for every such unknown
message identification.
</para><para>
There may be several reasons why this message may appear:
</para><para>
- The message ID has been mis-spelled in the local message file.
</para><para>
- The program outputting the message may not use that particular message
(e.g. it originates in a module not used by the program.)
</para><para>
- The local file was written for an earlier version of the BIND10 software
and the later version no longer generates that message.
</para><para>
Whatever the reason, there is no impact on the operation of BIND10.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_OPEN_OUTPUT_FAIL">
<term>LOG_OPEN_OUTPUT_FAIL unable to open %1 for output: %2</term>
<listitem><para>
Originating within the logging code, the program was not able to open
the specified output file for the reason given.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_PREFIX_EXTRA_ARGS">
<term>LOG_PREFIX_EXTRA_ARGS line %1: $PREFIX directive has too many arguments</term>
<listitem><para>
Within a message file, the $PREFIX directive takes a single argument,
a prefix to be added to the symbol names when a C++ file is created.
This error is generated when the compiler finds a $PREFIX directive with
more than one argument.
</para><para>
Note: the $PREFIX directive is deprecated and will be removed in a future
version of BIND10.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_PREFIX_INVALID_ARG">
<term>LOG_PREFIX_INVALID_ARG line %1: $PREFIX directive has an invalid argument ('%2')</term>
<listitem><para>
Within a message file, the $PREFIX directive takes a single argument,
a prefix to be added to the symbol names when a C++ file is created.
As such, it must adhere to restrictions on C++ symbol names (e.g. may
only contain alphanumeric characters or underscores, and may nor start
with a digit).  A $PREFIX directive was found with an argument (given
in the message) that violates those restictions.
</para><para>
Note: the $PREFIX directive is deprecated and will be removed in a future
version of BIND10.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_READING_LOCAL_FILE">
<term>LOG_READING_LOCAL_FILE reading local message file %1</term>
<listitem><para>
This is an informational message output by BIND10 when it starts to read
a local message file.  (A local message file may replace the text of
one of more messages; the ID of the message will not be changed though.)
</para></listitem>
</varlistentry>

<varlistentry id="LOG_READ_ERROR">
<term>LOG_READ_ERROR error reading from message file %1: %2</term>
<listitem><para>
The specified error was encountered reading from the named message file.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_UNRECOGNISED_DIRECTIVE">
<term>LOG_UNRECOGNISED_DIRECTIVE line %1: unrecognised directive '%2'</term>
<listitem><para>
Within a message file, a line starting with a dollar symbol was found
(indicating the presence of a directive) but the first word on the line
(shown in the message) was not recognised.
</para></listitem>
</varlistentry>

<varlistentry id="LOG_WRITE_ERROR">
<term>LOG_WRITE_ERROR error writing to %1: %2</term>
<listitem><para>
The specified error was encountered by the message compiler when writing
to the named output file.
</para></listitem>
</varlistentry>

<varlistentry id="NSAS_FIND_NS_ADDRESS">
<term>NSAS_FIND_NS_ADDRESS asking resolver to obtain A and AAAA records for %1</term>
<listitem><para>
A debug message issued when the NSAS (nameserver address store - part
of the resolver) is making a callback into the resolver to retrieve the
address records for the specified nameserver.
</para></listitem>
</varlistentry>

<varlistentry id="NSAS_FOUND_ADDRESS">
<term>NSAS_FOUND_ADDRESS found address %1 for %2</term>
<listitem><para>
A debug message issued when the NSAS (nameserver address store - part
of the resolver) has retrieved the given address for the specified
nameserver through an external query.
</para></listitem>
</varlistentry>

<varlistentry id="NSAS_INVALID_RESPONSE">
<term>NSAS_INVALID_RESPONSE queried for %1 but got invalid response</term>
<listitem><para>
The NSAS (nameserver address store - part of the resolver) made a query
for a RR for the specified nameserver but received an invalid response.
Either the success function was called without a DNS message or the
message was invalid on some way. (In the latter case, the error should
have been picked up elsewhere in the processing logic, hence the raising
of the error here.)
</para><para>
This message indicates an internal error in the NSAS.  Please raise a
bug report.
</para></listitem>
</varlistentry>

<varlistentry id="NSAS_LOOKUP_CANCEL">
<term>NSAS_LOOKUP_CANCEL lookup for zone %1 has been canceled</term>
<listitem><para>
A debug message issued when an NSAS (nameserver address store - part of
the resolver) lookup for a zone has been canceled.
</para></listitem>
</varlistentry>

<varlistentry id="NSAS_NS_LOOKUP_FAIL">
<term>NSAS_NS_LOOKUP_FAIL failed to lookup any %1 for %2</term>
<listitem><para>
A debug message issued when the NSAS (nameserver address store - part of
the resolver) has been unable to retrieve the specified resource record
for the specified nameserver.  This is not necessarily a problem - the
nameserver may be unreachable, in which case the NSAS will try other
nameservers in the zone.
</para></listitem>
</varlistentry>

<varlistentry id="NSAS_SEARCH_ZONE_NS">
<term>NSAS_SEARCH_ZONE_NS searching NSAS for nameservers for zone %1</term>
<listitem><para>
A debug message output when a call is made to the NSAS (nameserver
address store - part of the resolver) to obtain the nameservers for
the specified zone.
</para></listitem>
</varlistentry>

<varlistentry id="NSAS_UPDATE_RTT">
<term>NSAS_UPDATE_RTT update RTT for %1: was %2 ms, is now %3 ms</term>
<listitem><para>
A NSAS (nameserver address store - part of the resolver) debug message
reporting the update of a round-trip time (RTT) for a query made to the
specified nameserver.  The RTT has been updated using the value given
and the new RTT is displayed.  (The RTT is subject to a calculation that
damps out sudden changes.  As a result, the new RTT used by the NSAS in
future decisions of which nameserver to use is not necessarily equal to
the RTT reported.)
</para></listitem>
</varlistentry>

<varlistentry id="NSAS_WRONG_ANSWER">
<term>NSAS_WRONG_ANSWER queried for %1 RR of type/class %2/%3, received response %4/%5</term>
<listitem><para>
A NSAS (nameserver address store - part of the resolver) made a query for
a resource record of a particular type and class, but instead received
an answer with a different given type and class.
</para><para>
This message indicates an internal error in the NSAS.  Please raise a
bug report.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_ANSWER">
<term>RESLIB_ANSWER answer received in response to query for &lt;%1&gt;</term>
<listitem><para>
A debug message recording that an answer has been received to an upstream
query for the specified question.  Previous debug messages will have indicated
the server to which the question was sent.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_CNAME">
<term>RESLIB_CNAME CNAME received in response to query for &lt;%1&gt;</term>
<listitem><para>
A debug message recording that CNAME response has been received to an upstream
query for the specified question.  Previous debug messages will have indicated
the server to which the question was sent.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_DEEPEST">
<term>RESLIB_DEEPEST did not find &lt;%1&gt; in cache, deepest delegation found is %2</term>
<listitem><para>
A debug message, a cache lookup did not find the specified &lt;name, class,
type&gt; tuple in the cache; instead, the deepest delegation found is indicated.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_FOLLOW_CNAME">
<term>RESLIB_FOLLOW_CNAME following CNAME chain to &lt;%1&gt;</term>
<listitem><para>
A debug message, a CNAME response was received and another query is being issued
for the &lt;name, class, type&gt; tuple.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_LONG_CHAIN">
<term>RESLIB_LONG_CHAIN CNAME received in response to query for &lt;%1&gt;: CNAME chain length exceeded</term>
<listitem><para>
A debug message recording that a CNAME response has been received to an upstream
query for the specified question (Previous debug messages will have indicated
the server to which the question was sent).  However, receipt of this CNAME
has meant that the resolver has exceeded the CNAME chain limit (a CNAME chain
is where on CNAME points to another) and so an error is being returned.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_NO_NS_RRSET">
<term>RESLIB_NO_NS_RRSET no NS RRSet in referral response received to query for &lt;%1&gt;</term>
<listitem><para>
A debug message, this indicates that a response was received for the specified
query and was categorized as a referral.  However, the received message did
not contain any NS RRsets.  This may indicate a programming error in the
response classification code.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_NSAS_LOOKUP">
<term>RESLIB_NSAS_LOOKUP looking up nameserver for zone %1 in the NSAS</term>
<listitem><para>
A debug message, the RunningQuery object is querying the NSAS for the
nameservers for the specified zone.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_NXDOM_NXRR">
<term>RESLIB_NXDOM_NXRR NXDOMAIN/NXRRSET received in response to query for &lt;%1&gt;</term>
<listitem><para>
A debug message recording that either a NXDOMAIN or an NXRRSET response has
been received to an upstream query for the specified question.  Previous debug
messages will have indicated the server to which the question was sent.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_PROTOCOL">
<term>RESLIB_PROTOCOL protocol error in answer for %1:  %3</term>
<listitem><para>
A debug message indicating that a protocol error was received.  As there
are no retries left, an error will be reported.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_PROTOCOL_RETRY">
<term>RESLIB_PROTOCOL_RETRY protocol error in answer for %1: %2 (retries left: %3)</term>
<listitem><para>
A debug message indicating that a protocol error was received and that
the resolver is repeating the query to the same nameserver.  After this
repeated query, there will be the indicated number of retries left.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_RCODE_ERR">
<term>RESLIB_RCODE_ERR RCODE indicates error in response to query for &lt;%1&gt;</term>
<listitem><para>
A debug message, the response to the specified query indicated an error
that is not covered by a specific code path.  A SERVFAIL will be returned.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_RECQ_CACHE_FIND">
<term>RESLIB_RECQ_CACHE_FIND found &lt;%1&gt; in the cache (resolve() instance %2)</term>
<listitem><para>
This is a debug message and indicates that a RecursiveQuery object found the
the specified &lt;name, class, type&gt; tuple in the cache.  The instance number
at the end of the message indicates which of the two resolve() methods has
been called.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_RECQ_CACHE_NO_FIND">
<term>RESLIB_RECQ_CACHE_NO_FIND did not find &lt;%1&gt; in the cache, starting RunningQuery (resolve() instance %2)</term>
<listitem><para>
This is a debug message and indicates that the look in the cache made by the
RecursiveQuery::resolve() method did not find an answer, so a new RunningQuery
object has been created to resolve the question.  The instance number at
the end of the message indicates which of the two resolve() methods has
been called.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_REFERRAL">
<term>RESLIB_REFERRAL referral received in response to query for &lt;%1&gt;</term>
<listitem><para>
A debug message recording that a referral response has been received to an
upstream query for the specified question.  Previous debug messages will
have indicated the server to which the question was sent.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_REFER_ZONE">
<term>RESLIB_REFER_ZONE referred to zone %1</term>
<listitem><para>
A debug message indicating that the last referral message was to the specified
zone.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_RESOLVE">
<term>RESLIB_RESOLVE asked to resolve &lt;%1&gt; (resolve() instance %2)</term>
<listitem><para>
A debug message, the RecursiveQuery::resolve method has been called to resolve
the specified &lt;name, class, type&gt; tuple.  The first action will be to lookup
the specified tuple in the cache.  The instance number at the end of the
message indicates which of the two resolve() methods has been called.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_RRSET_FOUND">
<term>RESLIB_RRSET_FOUND found single RRset in the cache when querying for &lt;%1&gt; (resolve() instance %2)</term>
<listitem><para>
A debug message, indicating that when RecursiveQuery::resolve queried the
cache, a single RRset was found which was put in the answer.  The instance
number at the end of the message indicates which of the two resolve()
methods has been called.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_RTT">
<term>RESLIB_RTT round-trip time of last query calculated as %1 ms</term>
<listitem><para>
A debug message giving the round-trip time of the last query and response.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_RUNQ_CACHE_FIND">
<term>RESLIB_RUNQ_CACHE_FIND found &lt;%1&gt; in the cache</term>
<listitem><para>
This is a debug message and indicates that a RunningQuery object found
the specified &lt;name, class, type&gt; tuple in the cache.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_RUNQ_CACHE_LOOKUP">
<term>RESLIB_RUNQ_CACHE_LOOKUP looking up up &lt;%1&gt; in the cache</term>
<listitem><para>
This is a debug message and indicates that a RunningQuery object has made
a call to its doLookup() method to look up the specified &lt;name, class, type&gt;
tuple, the first action of which will be to examine the cache.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_RUNQ_FAIL">
<term>RESLIB_RUNQ_FAIL failure callback - nameservers are unreachable</term>
<listitem><para>
A debug message indicating that a RunningQuery's failure callback has been
called because all nameservers for the zone in question are unreachable.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_RUNQ_SUCCESS">
<term>RESLIB_RUNQ_SUCCESS success callback - sending query to %1</term>
<listitem><para>
A debug message indicating that a RunningQuery's success callback has been
called because a nameserver has been found, and that a query is being sent
to the specified nameserver.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_TEST_SERVER">
<term>RESLIB_TEST_SERVER setting test server to %1(%2)</term>
<listitem><para>
This is an internal debugging message and is only generated in unit tests.
It indicates that all upstream queries from the resolver are being routed to
the specified server, regardless of the address of the nameserver to which
the query would normally be routed.  As it should never be seen in normal
operation, it is a warning message instead of a debug message.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_TEST_UPSTREAM">
<term>RESLIB_TEST_UPSTREAM sending upstream query for &lt;%1&gt; to test server at %2</term>
<listitem><para>
This is a debug message and should only be seen in unit tests.  A query for
the specified &lt;name, class, type&gt; tuple is being sent to a test nameserver
whose address is given in the message.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_TIMEOUT">
<term>RESLIB_TIMEOUT query &lt;%1&gt; to %2 timed out</term>
<listitem><para>
A debug message indicating that the specified query has timed out and as
there are no retries left, an error will be reported.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_TIMEOUT_RETRY">
<term>RESLIB_TIMEOUT_RETRY query &lt;%1&gt; to %2 timed out, re-trying (retries left: %3)</term>
<listitem><para>
A debug message indicating that the specified query has timed out and that
the resolver is repeating the query to the same nameserver.  After this
repeated query, there will be the indicated number of retries left.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_TRUNCATED">
<term>RESLIB_TRUNCATED response to query for &lt;%1&gt; was truncated, re-querying over TCP</term>
<listitem><para>
A debug message, this indicates that the response to the specified query was
truncated and that the resolver will be re-querying over TCP.  There are
various reasons why responses may be truncated, so this message is normal and
gives no cause for concern.
</para></listitem>
</varlistentry>

<varlistentry id="RESLIB_UPSTREAM">
<term>RESLIB_UPSTREAM sending upstream query for &lt;%1&gt; to %2</term>
<listitem><para>
A debug message indicating that a query for the specified &lt;name, class, type&gt;
tuple is being sent to a nameserver whose address is given in the message.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_AXFR_TCP">
<term>RESOLVER_AXFR_TCP AXFR request received over TCP</term>
<listitem><para>
A debug message, the resolver received a NOTIFY message over TCP.  The server
cannot process it and will return an error message to the sender with the
RCODE set to NOTIMP.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_AXFR_UDP">
<term>RESOLVER_AXFR_UDP AXFR request received over UDP</term>
<listitem><para>
A debug message, the resolver received a NOTIFY message over UDP.  The server
cannot process it (and in any case, an AXFR request should be sent over TCP)
and will return an error message to the sender with the RCODE set to FORMERR.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_CLIENT_TIME_SMALL">
<term>RESOLVER_CLIENT_TIME_SMALL client timeout of %1 is too small</term>
<listitem><para>
An error indicating that the configuration value specified for the query
timeout is too small.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_CONFIG_CHANNEL">
<term>RESOLVER_CONFIG_CHANNEL configuration channel created</term>
<listitem><para>
A debug message, output when the resolver has successfully established a
connection to the configuration channel.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_CONFIG_ERROR">
<term>RESOLVER_CONFIG_ERROR error in configuration: %1</term>
<listitem><para>
An error was detected in a configuration update received by the resolver. This
may be in the format of the configuration message (in which case this is a
programming error) or it may be in the data supplied (in which case it is
a user error).  The reason for the error, given as a parameter in the message,
will give more details.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_CONFIG_LOADED">
<term>RESOLVER_CONFIG_LOADED configuration loaded</term>
<listitem><para>
A debug message, output when the resolver configuration has been successfully
loaded.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_CONFIG_UPDATED">
<term>RESOLVER_CONFIG_UPDATED configuration updated: %1</term>
<listitem><para>
A debug message, the configuration has been updated with the specified
information.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_CREATED">
<term>RESOLVER_CREATED main resolver object created</term>
<listitem><para>
A debug message, output when the Resolver() object has been created.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_DNS_MESSAGE_RECEIVED">
<term>RESOLVER_DNS_MESSAGE_RECEIVED DNS message received: %1</term>
<listitem><para>
A debug message, this always precedes some other logging message and is the
formatted contents of the DNS packet that the other message refers to.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_DNS_MESSAGE_SENT">
<term>RESOLVER_DNS_MESSAGE_SENT DNS message of %1 bytes sent: %2</term>
<listitem><para>
A debug message, this contains details of the response sent back to the querying
system.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_FAILED">
<term>RESOLVER_FAILED resolver failed, reason: %1</term>
<listitem><para>
This is an error message output when an unhandled exception is caught by the
resolver.  All it can do is to shut down.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_FORWARD_ADDRESS">
<term>RESOLVER_FORWARD_ADDRESS setting forward address %1(%2)</term>
<listitem><para>
This message may appear multiple times during startup, and it lists the
forward addresses used by the resolver when running in forwarding mode.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_FORWARD_QUERY">
<term>RESOLVER_FORWARD_QUERY processing forward query</term>
<listitem><para>
The received query has passed all checks and is being forwarded to upstream
servers.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_HEADER_ERROR">
<term>RESOLVER_HEADER_ERROR message received, exception when processing header: %1</term>
<listitem><para>
A debug message noting that an exception occurred during the processing of
a received packet.  The packet has been dropped.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_IXFR">
<term>RESOLVER_IXFR IXFR request received</term>
<listitem><para>
The resolver received a NOTIFY message over TCP.  The server cannot process it
and will return an error message to the sender with the RCODE set to NOTIMP.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_LOOKUP_TIME_SMALL">
<term>RESOLVER_LOOKUP_TIME_SMALL lookup timeout of %1 is too small</term>
<listitem><para>
An error indicating that the configuration value specified for the lookup
timeout is too small.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_MESSAGE_ERROR">
<term>RESOLVER_MESSAGE_ERROR error parsing received message: %1 - returning %2</term>
<listitem><para>
A debug message noting that the resolver received a message and the
parsing of the body of the message failed due to some error (although
the parsing of the header succeeded).  The message parameters give a
textual description of the problem and the RCODE returned.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_NEGATIVE_RETRIES">
<term>RESOLVER_NEGATIVE_RETRIES negative number of retries (%1) specified in the configuration</term>
<listitem><para>
An error message indicating that the resolver configuration has specified a
negative retry count.  Only zero or positive values are valid.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_NON_IN_PACKET">
<term>RESOLVER_NON_IN_PACKET non-IN class request received, returning REFUSED message</term>
<listitem><para>
A debug message, the resolver has received a DNS packet that was not IN class.
The resolver cannot handle such packets, so is returning a REFUSED response to
the sender.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_NORMAL_QUERY">
<term>RESOLVER_NORMAL_QUERY processing normal query</term>
<listitem><para>
The received query has passed all checks and is being processed by the resolver.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_NOTIFY_RECEIVED">
<term>RESOLVER_NOTIFY_RECEIVED NOTIFY arrived but server is not authoritative</term>
<listitem><para>
The resolver received a NOTIFY message.  As the server is not authoritative it
cannot process it, so it returns an error message to the sender with the RCODE
set to NOTAUTH.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_NOT_ONE_QUESTION">
<term>RESOLVER_NOT_ONE_QUESTION query contained %1 questions, exactly one question was expected</term>
<listitem><para>
A debug message, the resolver received a query that contained the number of
entires in the question section detailed in the message.  This is a malformed
message, as a DNS query must contain only one question.  The resolver will
return a message to the sender with the RCODE set to FORMERR.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_NO_ROOT_ADDRESS">
<term>RESOLVER_NO_ROOT_ADDRESS no root addresses available</term>
<listitem><para>
A warning message during startup, indicates that no root addresses have been
set.  This may be because the resolver will get them from a priming query.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_PARSE_ERROR">
<term>RESOLVER_PARSE_ERROR error parsing received message: %1 - returning %2</term>
<listitem><para>
A debug message noting that the resolver received a message and the parsing
of the body of the message failed due to some non-protocol related reason
(although the parsing of the header succeeded).  The message parameters give
a textual description of the problem and the RCODE returned.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_PRINT_COMMAND">
<term>RESOLVER_PRINT_COMMAND print message command, arguments are: %1</term>
<listitem><para>
This message is logged when a "print_message" command is received over the
command channel.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_PROTOCOL_ERROR">
<term>RESOLVER_PROTOCOL_ERROR protocol error parsing received message: %1 - returning %2</term>
<listitem><para>
A debug message noting that the resolver received a message and the parsing
of the body of the message failed due to some protocol error (although the
parsing of the header succeeded).  The message parameters give a textual
description of the problem and the RCODE returned.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_QUERY_ACCEPTED">
<term>RESOLVER_QUERY_ACCEPTED query accepted: '%1/%2/%3' from %4</term>
<listitem><para>
A debug message that indicates an incoming query is accepted in terms of
the query ACL.  The log message shows the query in the form of
&lt;query name&gt;/&lt;query type&gt;/&lt;query class&gt;, and the client that sends the
query in the form of &lt;Source IP address&gt;#&lt;source port&gt;.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_QUERY_DROPPED">
<term>RESOLVER_QUERY_DROPPED query dropped: '%1/%2/%3' from %4</term>
<listitem><para>
An informational message that indicates an incoming query is dropped
in terms of the query ACL.  Unlike the RESOLVER_QUERY_REJECTED
case, the server does not return any response.  The log message
shows the query in the form of &lt;query name&gt;/&lt;query type&gt;/&lt;query
class&gt;, and the client that sends the query in the form of &lt;Source
IP address&gt;#&lt;source port&gt;.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_QUERY_REJECTED">
<term>RESOLVER_QUERY_REJECTED query rejected: '%1/%2/%3' from %4</term>
<listitem><para>
An informational message that indicates an incoming query is rejected
in terms of the query ACL.  This results in a response with an RCODE of
REFUSED. The log message shows the query in the form of &lt;query
name&gt;/&lt;query type&gt;/&lt;query class&gt;, and the client that sends the
query in the form of &lt;Source IP address&gt;#&lt;source port&gt;.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_QUERY_SETUP">
<term>RESOLVER_QUERY_SETUP query setup</term>
<listitem><para>
A debug message noting that the resolver is creating a RecursiveQuery object.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_QUERY_SHUTDOWN">
<term>RESOLVER_QUERY_SHUTDOWN query shutdown</term>
<listitem><para>
A debug message noting that the resolver is destroying a RecursiveQuery object.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_QUERY_TIME_SMALL">
<term>RESOLVER_QUERY_TIME_SMALL query timeout of %1 is too small</term>
<listitem><para>
An error indicating that the configuration value specified for the query
timeout is too small.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_RECEIVED_MESSAGE">
<term>RESOLVER_RECEIVED_MESSAGE resolver has received a DNS message</term>
<listitem><para>
A debug message indicating that the resolver has received a message.  Depending
on the debug settings, subsequent log output will indicate the nature of the
message.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_RECURSIVE">
<term>RESOLVER_RECURSIVE running in recursive mode</term>
<listitem><para>
This is an informational message that appears at startup noting that the
resolver is running in recursive mode.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_SERVICE_CREATED">
<term>RESOLVER_SERVICE_CREATED service object created</term>
<listitem><para>
A debug message, output when the main service object (which handles the
received queries) is created.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_SET_PARAMS">
<term>RESOLVER_SET_PARAMS query timeout: %1, client timeout: %2, lookup timeout: %3, retry count: %4</term>
<listitem><para>
A debug message, lists the parameters being set for the resolver.  These are:
query timeout: the timeout (in ms) used for queries originated by the resolver
to upstream servers.  Client timeout: the interval to resolver a query by
a client: after this time, the resolver sends back a SERVFAIL to the client
whilst continuing to resolver the query. Lookup timeout: the time at which the
resolver gives up trying to resolve a query.  Retry count: the number of times
the resolver will retry a query to an upstream server if it gets a timeout.
</para><para>
The client and lookup timeouts require a bit more explanation. The
resolution of the client query might require a large number of queries to
upstream nameservers.  Even if none of these queries timeout, the total time
taken to perform all the queries may exceed the client timeout.  When this
happens, a SERVFAIL is returned to the client, but the resolver continues
with the resolution process. Data received is added to the cache.  However,
there comes a time - the lookup timeout - when even the resolver gives up.
At this point it will wait for pending upstream queries to complete or
timeout and drop the query.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_SET_QUERY_ACL">
<term>RESOLVER_SET_QUERY_ACL query ACL is configured</term>
<listitem><para>
A debug message that appears when a new query ACL is configured for the
resolver.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_SET_ROOT_ADDRESS">
<term>RESOLVER_SET_ROOT_ADDRESS setting root address %1(%2)</term>
<listitem><para>
This message may appear multiple times during startup; it lists the root
addresses used by the resolver.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_SHUTDOWN">
<term>RESOLVER_SHUTDOWN resolver shutdown complete</term>
<listitem><para>
This information message is output when the resolver has shut down.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_STARTED">
<term>RESOLVER_STARTED resolver started</term>
<listitem><para>
This informational message is output by the resolver when all initialization
has been completed and it is entering its main loop.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_STARTING">
<term>RESOLVER_STARTING starting resolver with command line '%1'</term>
<listitem><para>
An informational message, this is output when the resolver starts up.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_UNEXPECTED_RESPONSE">
<term>RESOLVER_UNEXPECTED_RESPONSE received unexpected response, ignoring</term>
<listitem><para>
A debug message noting that the server has received a response instead of a
query and is ignoring it.
</para></listitem>
</varlistentry>

<varlistentry id="RESOLVER_UNSUPPORTED_OPCODE">
<term>RESOLVER_UNSUPPORTED_OPCODE opcode %1 not supported by the resolver</term>
<listitem><para>
A debug message, the resolver received a message with an unsupported opcode
(it can only process QUERY opcodes).  It will return a message to the sender
with the RCODE set to NOTIMP.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_AXFR_DATABASE_FAILURE">
<term>XFRIN_AXFR_DATABASE_FAILURE AXFR transfer of zone %1 failed: %2</term>
<listitem><para>
The AXFR transfer for the given zone has failed due to a database problem.
The error is shown in the log message.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_AXFR_INTERNAL_FAILURE">
<term>XFRIN_AXFR_INTERNAL_FAILURE AXFR transfer of zone %1 failed: %2</term>
<listitem><para>
The AXFR transfer for the given zone has failed due to an internal
problem in the bind10 python wrapper library.
The error is shown in the log message.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_AXFR_TRANSFER_FAILURE">
<term>XFRIN_AXFR_TRANSFER_FAILURE AXFR transfer of zone %1 failed: %2</term>
<listitem><para>
The AXFR transfer for the given zone has failed due to a protocol error.
The error is shown in the log message.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_AXFR_TRANSFER_STARTED">
<term>XFRIN_AXFR_TRANSFER_STARTED AXFR transfer of zone %1 started</term>
<listitem><para>
A connection to the master server has been made, the serial value in
the SOA record has been checked, and a zone transfer has been started.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_AXFR_TRANSFER_SUCCESS">
<term>XFRIN_AXFR_TRANSFER_SUCCESS AXFR transfer of zone %1 succeeded</term>
<listitem><para>
The AXFR transfer of the given zone was successfully completed.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_BAD_MASTER_ADDR_FORMAT">
<term>XFRIN_BAD_MASTER_ADDR_FORMAT bad format for master address: %1</term>
<listitem><para>
The given master address is not a valid IP address.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_BAD_MASTER_PORT_FORMAT">
<term>XFRIN_BAD_MASTER_PORT_FORMAT bad format for master port: %1</term>
<listitem><para>
The master port as read from the configuration is not a valid port number.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_BAD_TSIG_KEY_STRING">
<term>XFRIN_BAD_TSIG_KEY_STRING bad TSIG key string: %1</term>
<listitem><para>
The TSIG key string as read from the configuration does not represent
a valid TSIG key.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_BAD_ZONE_CLASS">
<term>XFRIN_BAD_ZONE_CLASS Invalid zone class: %1</term>
<listitem><para>
The zone class as read from the configuration is not a valid DNS class.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_CC_SESSION_ERROR">
<term>XFRIN_CC_SESSION_ERROR error reading from cc channel: %1</term>
<listitem><para>
There was a problem reading from the command and control channel. The
most likely cause is that xfrin the msgq daemon is not running.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_COMMAND_ERROR">
<term>XFRIN_COMMAND_ERROR error while executing command '%1': %2</term>
<listitem><para>
There was an error while the given command was being processed. The
error is given in the log message.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_CONNECT_MASTER">
<term>XFRIN_CONNECT_MASTER error connecting to master at %1: %2</term>
<listitem><para>
There was an error opening a connection to the master. The error is
shown in the log message.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_IMPORT_DNS">
<term>XFRIN_IMPORT_DNS error importing python DNS module: %1</term>
<listitem><para>
There was an error importing the python DNS module pydnspp. The most
likely cause is a PYTHONPATH problem.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_MSGQ_SEND_ERROR">
<term>XFRIN_MSGQ_SEND_ERROR error while contacting %1 and %2</term>
<listitem><para>
There was a problem sending a message to the xfrout module or the
zone manager. This most likely means that the msgq daemon has quit or
was killed.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_MSGQ_SEND_ERROR_ZONE_MANAGER">
<term>XFRIN_MSGQ_SEND_ERROR_ZONE_MANAGER error while contacting %1</term>
<listitem><para>
There was a problem sending a message to the zone manager. This most
likely means that the msgq daemon has quit or was killed.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_RETRANSFER_UNKNOWN_ZONE">
<term>XFRIN_RETRANSFER_UNKNOWN_ZONE got notification to retransfer unknown zone %1</term>
<listitem><para>
There was an internal command to retransfer the given zone, but the
zone is not known to the system. This may indicate that the configuration
for xfrin is incomplete, or there was a typographical error in the
zone name in the configuration.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_STARTING">
<term>XFRIN_STARTING starting resolver with command line '%1'</term>
<listitem><para>
An informational message, this is output when the resolver starts up.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_STOPPED_BY_KEYBOARD">
<term>XFRIN_STOPPED_BY_KEYBOARD keyboard interrupt, shutting down</term>
<listitem><para>
There was a keyboard interrupt signal to stop the xfrin daemon. The
daemon will now shut down.
</para></listitem>
</varlistentry>

<varlistentry id="XFRIN_UNKNOWN_ERROR">
<term>XFRIN_UNKNOWN_ERROR unknown error: %1</term>
<listitem><para>
An uncaught exception was raised while running the xfrin daemon. The
exception message is printed in the log message.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_AXFR_TRANSFER_DONE">
<term>XFROUT_AXFR_TRANSFER_DONE transfer of %1/%2 complete</term>
<listitem><para>
The transfer of the given zone has been completed successfully, or was
aborted due to a shutdown event.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_AXFR_TRANSFER_ERROR">
<term>XFROUT_AXFR_TRANSFER_ERROR error transferring zone %1/%2: %3</term>
<listitem><para>
An uncaught exception was encountered while sending the response to
an AXFR query. The error message of the exception is included in the
log message, but this error most likely points to incomplete exception
handling in the code.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_AXFR_TRANSFER_FAILED">
<term>XFROUT_AXFR_TRANSFER_FAILED transfer of %1/%2 failed, rcode: %3</term>
<listitem><para>
A transfer out for the given zone failed. An error response is sent
to the client. The given rcode is the rcode that is set in the error
response. This is either NOTAUTH (we are not authoritative for the
zone), SERVFAIL (our internal database is missing the SOA record for
the zone), or REFUSED (the limit of simultaneous outgoing AXFR
transfers, as specified by the configuration value
Xfrout/max_transfers_out, has been reached).
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_AXFR_TRANSFER_STARTED">
<term>XFROUT_AXFR_TRANSFER_STARTED transfer of zone %1/%2 has started</term>
<listitem><para>
A transfer out of the given zone has started.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_BAD_TSIG_KEY_STRING">
<term>XFROUT_BAD_TSIG_KEY_STRING bad TSIG key string: %1</term>
<listitem><para>
The TSIG key string as read from the configuration does not represent
a valid TSIG key.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_CC_SESSION_ERROR">
<term>XFROUT_CC_SESSION_ERROR error reading from cc channel: %1</term>
<listitem><para>
There was a problem reading from the command and control channel. The
most likely cause is that the msgq daemon is not running.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_CC_SESSION_TIMEOUT_ERROR">
<term>XFROUT_CC_SESSION_TIMEOUT_ERROR timeout waiting for cc response</term>
<listitem><para>
There was a problem reading a response from antoher module over the
command and control channel. The most likely cause is that the
configuration manager b10-cfgmgr is not running.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_FETCH_REQUEST_ERROR">
<term>XFROUT_FETCH_REQUEST_ERROR socket error while fetching a request from the auth daemon</term>
<listitem><para>
There was a socket error while contacting the b10-auth daemon to
fetch a transfer request. The auth daemon may have shutdown.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_HANDLE_QUERY_ERROR">
<term>XFROUT_HANDLE_QUERY_ERROR error while handling query: %1</term>
<listitem><para>
There was a general error handling an xfrout query. The error is shown
in the message. In principle this error should not appear, and points
to an oversight catching exceptions in the right place. However, to
ensure the daemon keeps running, this error is caught and reported.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_IMPORT">
<term>XFROUT_IMPORT error importing python module: %1</term>
<listitem><para>
There was an error importing a python module. One of the modules needed
by xfrout could not be found. This suggests that either some libraries
are missing on the system, or the PYTHONPATH variable is not correct.
The specific place where this library needs to be depends on your
system and your specific installation.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_NEW_CONFIG">
<term>XFROUT_NEW_CONFIG Update xfrout configuration</term>
<listitem><para>
New configuration settings have been sent from the configuration
manager. The xfrout daemon will now apply them.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_NEW_CONFIG_DONE">
<term>XFROUT_NEW_CONFIG_DONE Update xfrout configuration done</term>
<listitem><para>
The xfrout daemon is now done reading the new configuration settings
received from the configuration manager.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_NOTIFY_COMMAND">
<term>XFROUT_NOTIFY_COMMAND received command to send notifies for %1/%2</term>
<listitem><para>
The xfrout daemon received a command on the command channel that
NOTIFY packets should be sent for the given zone.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_PARSE_QUERY_ERROR">
<term>XFROUT_PARSE_QUERY_ERROR error parsing query: %1</term>
<listitem><para>
There was a parse error while reading an incoming query. The parse
error is shown in the log message. A remote client sent a packet we
do not understand or support. The xfrout request will be ignored.
In general, this should only occur for unexpected problems like
memory allocation failures, as the query should already have been
parsed by the b10-auth daemon, before it was passed here.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_PROCESS_REQUEST_ERROR">
<term>XFROUT_PROCESS_REQUEST_ERROR error processing transfer request: %2</term>
<listitem><para>
There was an error processing a transfer request. The error is included
in the log message, but at this point no specific information other
than that could be given. This points to incomplete exception handling
in the code.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_RECEIVED_SHUTDOWN_COMMAND">
<term>XFROUT_RECEIVED_SHUTDOWN_COMMAND shutdown command received</term>
<listitem><para>
The xfrout daemon received a shutdown command from the command channel
and will now shut down.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_RECEIVE_FILE_DESCRIPTOR_ERROR">
<term>XFROUT_RECEIVE_FILE_DESCRIPTOR_ERROR error receiving the file descriptor for an XFR connection</term>
<listitem><para>
There was an error receiving the file descriptor for the transfer
request. Normally, the request is received by b10-auth, and passed on
to the xfrout daemon, so it can answer directly. However, there was a
problem receiving this file descriptor. The request will be ignored.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_REMOVE_OLD_UNIX_SOCKET_FILE_ERROR">
<term>XFROUT_REMOVE_OLD_UNIX_SOCKET_FILE_ERROR error removing unix socket file %1: %2</term>
<listitem><para>
The unix socket file xfrout needs for contact with the auth daemon
already exists, and needs to be removed first, but there is a problem
removing it. It is likely that we do not have permission to remove
this file. The specific error is show in the log message. The xfrout
daemon will shut down.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_REMOVE_UNIX_SOCKET_FILE_ERROR">
<term>XFROUT_REMOVE_UNIX_SOCKET_FILE_ERROR error clearing unix socket file %1: %2</term>
<listitem><para>
When shutting down, the xfrout daemon tried to clear the unix socket
file used for communication with the auth daemon. It failed to remove
the file. The reason for the failure is given in the error message.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_SOCKET_SELECT_ERROR">
<term>XFROUT_SOCKET_SELECT_ERROR error while calling select() on request socket: %1</term>
<listitem><para>
There was an error while calling select() on the socket that informs
the xfrout daemon that a new xfrout request has arrived. This should
be a result of rare local error such as memory allocation failure and
shouldn't happen under normal conditions. The error is included in the
log message.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_STOPPED_BY_KEYBOARD">
<term>XFROUT_STOPPED_BY_KEYBOARD keyboard interrupt, shutting down</term>
<listitem><para>
There was a keyboard interrupt signal to stop the xfrout daemon. The
daemon will now shut down.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_STOPPING">
<term>XFROUT_STOPPING the xfrout daemon is shutting down</term>
<listitem><para>
The current transfer is aborted, as the xfrout daemon is shutting down.
</para></listitem>
</varlistentry>

<varlistentry id="XFROUT_UNIX_SOCKET_FILE_IN_USE">
<term>XFROUT_UNIX_SOCKET_FILE_IN_USE another xfrout process seems to be using the unix socket file %1</term>
<listitem><para>
While starting up, the xfrout daemon tried to clear the unix domain
socket needed for contacting the b10-auth daemon to pass requests
on, but the file is in use. The most likely cause is that another
xfrout daemon process is still running. This xfrout daemon (the one
printing this message) will not start.
</para></listitem>
</varlistentry>
      </variablelist>
    </para>
  </chapter>
</book>
